### DISCUSSION TITLE
Generate HealthyHostCount metric for retrieved NetworkTargetGroup

### DISCUSSION THREAD

**Discussion by marcinpz** [aws/aws-cdk#21943] [Q&A]
Hi,
I see there is method `NetworkTargetGroup.metricHealthyHostCount()` to generate `HealthyHostCount` metric but when I use `elbv2.NetworkTargetGroup.fromTargetGroupAttributes()` to obtain existing instance then it gives object of type `INetworkTargetGroup` which does not implement `metricHealthyHostCount`. I can use sth like this:
```
new cloudwatch.Metric({
            // https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-cloudwatch-metrics.html
            namespace: 'AWS/ApplicationELB',
            metricName: 'HealthyHostCount',
 ...
```
but previous method looks better/is shorter/don't need to specify some params.
And sth like this (casting) does not work:
```
    const metric = (targetGroup as elbv2.NetworkTargetGroup).metricHealthyHostCount();
                                                                 ^
TypeError: targetGroup.metricHealthyHostCount is not a function
```

**Comment by pahud** [comment]
[metricHealthHostCount()](https://github.com/aws/aws-cdk/blob/cea1039e3664fdfa89c6f00cdaeb1a0185a12678/packages/%40aws-cdk/aws-elasticloadbalancingv2/lib/nlb/network-target-group.ts#L125) is actually a construct method of [NetworkTargetGroup](https://github.com/aws/aws-cdk/blob/cea1039e3664fdfa89c6f00cdaeb1a0185a12678/packages/%40aws-cdk/aws-elasticloadbalancingv2/lib/nlb/network-target-group.ts#L59) L2 construct class and only available when you create a new `NetworkTargetGroup` object.

**Reply by marcinpz** [reply]
I see but don't understand why?! Why I cant generate metric for existing NTG easy way??
